lib/modules/versioning/composer/index.spec.ts

```diff
@@ -91,6 +101,7 @@ describe('modules/versioning/composer/index', () => {
     ${['v0.4.0', 'v0.5.0', 'v4.0.0', 'v4.2.0', 'v5.0.0']}                                      | ${'~4'}      | ${'v4.2.0'}
     ${['0.4.0', '0.5.0', '4.0.0', '4.2.0', '5.0.0']}                                           | ${'~0.4'}    | ${'0.5.0'}
     ${['0.4.0', '0.5.0', '4.0.0-beta1', '4.0.0-beta2', '4.2.0-beta1', '4.2.0-beta2', '5.0.0']} | ${'~4@beta'} | ${'4.0.0-beta2'}
+    ${['0.4.0', '0.5.0', '4.0.0-p1', '4.0.0-p2', '4.2.0-p1', '4.2.0-p2', '5.0.0']}             | ${'~4@p'}    | ${'4.0.0-p2'}
```

Comment:
why doesn't it update to `4.2.0-p2`?

Reply:
this test is analogue to the line above - `minSatisfyingVersion` - it seems to be correct taking the lowest possible patch version matching 4@p. Did I misunderstand the purpose of that function?

Reply:
I think `getSatisfyingVersion()` should return the highest satisfying

Reply:
so it seems there is a bug, probably also there before your changes